src/crypto.rs

```diff
@@ -69,6 +80,51 @@ pub(crate) fn pkey_pub_from_priv(
     }
 }
 
+pub(crate) fn generate_x509(key: &PKey<Private>, uuid: &str) -> Result<X509> {
+    let mut name = X509Name::builder()?;
+    name.append_entry_by_nid(Nid::COMMONNAME, uuid)?;
+    let name = name.build();
+
+    let valid_from = Asn1Time::days_from_now(0)?;
+    let valid_to = Asn1Time::days_from_now(356)?;
+
+    let mut builder = X509::builder()?;
+    builder.set_version(2)?;
+    builder.set_subject_name(&name)?;
+    builder.set_issuer_name(&name)?;
+    builder.set_not_before(&valid_from)?;
+    builder.set_not_after(&valid_to)?;
+    builder.set_pubkey(key)?;
```

Comment:
Although this use-case is a bit special, I wonder if it would make sense to add some common restrictions such as (E)KU, SAN, etc?

Reply:
SAN does not make sense because we ignore it on the server side. We do this because we don't know on which IP or address we use to connect to the agent beforehand.

What attributes would you set for (E)KU?